### PR TITLE
Fixes for features + new configs

### PR DESCRIPTION
**Pull request by Alexandra-Myers**
The PR I made after giving feedback on the discord

### CODE REVIEW

File: src/main/java/com/infamous/combat_next/util/CombatUtil.java [hunk 1]
```diff
@@ -159,8 +162,8 @@ public static void attackEmpty(Player player) {
     
     public static void resetAttackStrengthTicker(Player player, boolean miss, boolean allowDefault){
         if(miss && MeleeCombatConfigs.getAttackMissReducedCooldown().get()){
-            LivingEntityAccessor accessor = (LivingEntityAccessor) player;
-            accessor.setAttackStrengthTicker(Math.max(0, (int) (player.getCurrentItemAttackStrengthDelay() - MeleeCombatConfigs.getAttackMissCooldownTicks().get())));
+            player.resetAttackStrengthTicker();
```
Comment (Thelnfamous1) [inline src/main/java/com/infamous/combat_next/util/CombatUtil.java:165]: Does the attack strength ticker still need to be reset here?

File: src/main/java/com/infamous/combat_next/util/CombatUtil.java [hunk 2]
```diff
@@ -170,7 +173,7 @@ private static void sweepAttack(Player player){
         float attackDamage = (float)player.getAttributeValue(Attributes.ATTACK_DAMAGE);
         float damageBonus = EnchantmentHelper.getDamageBonus(player.getMainHandItem(), MobType.UNDEFINED);
 
-        float attackStrengthScale = player.getAttackStrengthScale(0.5F);
+        float attackStrengthScale = MeleeCombatConfigs.getAttackCooldownImpactOnDamage().get() ? player.getAttackStrengthScale(0.5F) : 1;
```
Comment (Thelnfamous1) [inline src/main/java/com/infamous/combat_next/util/CombatUtil.java:176]: This config should be named in a way that specifies it is for missed attack sweep damage (however you can shorten that).

File: src/main/java/com/infamous/combat_next/util/CombatUtil.java [hunk 3]
```diff
@@ -218,16 +221,44 @@ private static AABB getSweepHitBox(Player player, ItemStack weapon) {
         return weapon.getSweepHitBox(player, player).move(xShift, 0.0, zShift);
     }
 
+    public static boolean isAttackAvailable(Player player, float partialTick) {
+        if (player.getAttackStrengthScale(partialTick) < 1.0F) {
+            LivingEntityAccessor livingEntityAccessor = (LivingEntityAccessor) player;
+            return PlayerCombat.cast(player).getMissedAttackRecovery() > 0 && livingEntityAccessor.getAttackStrengthTicker() + partialTick > PlayerCombat.cast(player).getMissedAttackRecovery();
```
Comment (Thelnfamous1) [inline src/main/java/com/infamous/combat_next/util/CombatUtil.java]: Does the missed attack recovery need to have an impact if it is 0 or less?

File: src/main/java/com/infamous/combat_next/mixin/PlayerMixin.java [hunk 4]
```diff
@@ -58,6 +60,11 @@ private float bypassZeroDamage(float constant, DamageSource source, float damage
         }
     }
 
+    @ModifyExpressionValue(method = "attack", at = @At(value = "INVOKE", target = "Lnet/minecraft/world/entity/player/Player;getAttackStrengthScale(F)F", ordinal = 0))
+    public float nullTime(float original) {
+        return MeleeCombatConfigs.getAttackCooldownImpactOnDamage().get() ? original : 1;
```
Comment (Thelnfamous1) [inline src/main/java/com/infamous/combat_next/mixin/PlayerMixin.java:65]: So if you saw the comment I made in CombatUtil about this, feel free to ignore it if it is indeed meant to be used for multiple damage calculations apart from sweep.

File: src/main/java/com/infamous/combat_next/config/GeneralCombatConfigs.java [hunk 5]
```diff
@@ -37,6 +39,16 @@ Adjusts the time in ticks (1/20 seconds) it takes to consume a drink when used.
...
+            hitboxAdjustmentType = b
+                    .comment("""
+                            Which type of hitbox adjustment should be used.
+                            Base CN hitbox increase expands all dimensions to the minimum value.
+                            CTS hitbox increase only applies if the largest dimension of the entity is less than the minimum.
+                            If CTS does apply to the entity, it inflates all dimensions by the value (minimum - largestDim).
+                            For vanilla, this value is OFF.
+                            For CTS, this value is CTS.
+                            """)
+                    .defineEnum("hitbox_adjustment_type", HitboxInflationType.CN);
```
Comment (Thelnfamous1) [inline src/main/java/com/infamous/combat_next/config/GeneralCombatConfigs.java]: Make the default HitboxInflationType.CTS as per our discussion

File: src/main/java/com/infamous/combat_next/config/MeleeCombatConfigs.java [hunk 6]
```diff
@@ -53,6 +63,18 @@ static void createServerConfigs(ForgeConfigSpec.Builder builder) {
                             For vanilla, this value is false.
                             """)
                     .define("attack_during_cooldown_prevented", true);
+            attackGracePeriod = b
+                    .comment("""
+                            Toggles the "grace period" where if you attack before 100% when you shouldn't be able to, but are between 80% and 100%, the attack is delayed until one tick after 100%.
```
Comment (Thelnfamous1) [inline src/main/java/com/infamous/combat_next/config/MeleeCombatConfigs.java]: Change the 80% value here to specify it is the value in the following attackGracePeriodTime config

**Review by Thelnfamous1 — CHANGES_REQUESTED** [review]
Please review the comments and make changes as needed.

File: src/main/java/com/infamous/combat_next/util/CombatUtil.java (hunk 1 above)
Comment (Alexandra-Myers) [reply src/main/java/com/infamous/combat_next/util/CombatUtil.java:165]: yes

File: src/main/java/com/infamous/combat_next/util/CombatUtil.java (hunk 2 above)
Comment (Alexandra-Myers) [reply src/main/java/com/infamous/combat_next/util/CombatUtil.java:176]: it isn't, though, it is also used for a standard attack where you do not miss

File: src/main/java/com/infamous/combat_next/util/CombatUtil.java (hunk 3 above)
Comment (Alexandra-Myers) [reply src/main/java/com/infamous/combat_next/util/CombatUtil.java]: not if it is less than 0, but maybe if it is 0

File: src/main/java/com/infamous/combat_next/mixin/PlayerMixin.java (hunk 4 above)
Comment (Alexandra-Myers) [reply src/main/java/com/infamous/combat_next/mixin/PlayerMixin.java:65]: yes, it is. this mixin shows that

File: src/main/java/com/infamous/combat_next/config/GeneralCombatConfigs.java (hunk 5 above)
Comment (Alexandra-Myers) [reply src/main/java/com/infamous/combat_next/config/GeneralCombatConfigs.java]: oh I thought you meant default CN

File: src/main/java/com/infamous/combat_next/config/MeleeCombatConfigs.java (hunk 6 above)
Comment (Alexandra-Myers) [reply src/main/java/com/infamous/combat_next/config/MeleeCombatConfigs.java]: makes sense

Comment (Alexandra-Myers) [reply src/main/java/com/infamous/combat_next/config/MeleeCombatConfigs.java]: wrote it before adding the time config

File: src/main/java/com/infamous/combat_next/util/CombatUtil.java (hunk 3 above)
Comment (Alexandra-Myers) [reply src/main/java/com/infamous/combat_next/util/CombatUtil.java]: oh wait, the default value missedAttackRecovery gets set to is 0, and it is reset to zero, so yes

Comment (Alexandra-Myers) [reply src/main/java/com/infamous/combat_next/util/CombatUtil.java]: I made the default value -1 so it can be set as 0 without not working

Comment (Alexandra-Myers) [reply src/main/java/com/infamous/combat_next/util/CombatUtil.java]: as in, default value for the variable used to know if an attack is a missed attack and how early it lets you, not the config default

**Comment by Alexandra-Myers** [comment]
In case you didn't notice, I have resolved all of your changes

**Review by Thelnfamous1 — APPROVED** [review]

**Pull request merged**